### ISSUE TITLE
Blocking  issue

### ISSUE THREAD

**Issue by thundergreen** [tgalal/yowsup#1516]
I wanted to ask if there's already a workaround for the blocking issue and if this is focused by you guys.don't know if it makes sense to investigate other issues before this one.

**Comment by t0rr3sp3dr0** [comment]
Agreed

**Comment by t0rr3sp3dr0** [comment]
All my numbers were blocked last Friday at 12h42 UTC and I don't know what to do. I think we need to focus on this problem, there's no sense in keep investigating other problems if yowsup is now unusable due to this problem.

**Comment by jegue** [comment]
use telegram

**Comment by thundergreen** [comment]
If this would heal the global Whatsapp Problem that most of the people use
wa ...i would rather use telegram .but there's a sense having yowsup i
think ;) i already tried to use a virtual number to register to wa as I
only have one number but failed. I think wa already git this trick and
built in a stop

Thorsten Fröhlich
Am 17.04.2016 20:13 schrieb "jegue" <notifications@github.com>:

**Comment by ronromer** [comment]
+1,  blocked virtual and real number .. Not tried anymore until workaround .. Also blocked with chat-api . I'm not sendin spam or any bad behavior

**Comment by assegaf** [comment]
@ronromer they already  implemented WA20 Protocol for 1 month so basically its easy to detect which is not using WA20 protocol will be banned 100% .. no matter what you do. We should move to protocol WA20 to avoid detection for using non-official WA Client.

**Comment by mvzeppelin** [comment]
I read on-post:  #1514 that WA20 protocolo is not supported on yowsup yet.
Is there anyone having successful sending messages since last friday? All my numbers was banned.

**Comment by tgalal** [comment]
@assegaf no, using non wa20 protocol at the moment doesn't mean you will be 100%
@mvzeppelin  use wa16 branch for now

**Comment by assegaf** [comment]
ah yes, just noticed that using whatsapp android 2.12.4XX still working, so WA16 still ok, so its just a matter of time until they block all WA16 protocol version.

**Comment by t0rr3sp3dr0** [comment]
Is there some workaround to register new numbers in yowsup?

**Comment by ilkerkopan** [comment]
Just want to share my baned number story :). I was using yowsup for test purposes. just sending from my own phone to yowsup client and getting back a message. After using a day (just sending 7 messages to a number in contacts), this morning I find out that yowsup client number was banned. In Turkey there is bank using whatsapp client bot for currency values, strangely it is working not getting banned, I guess they made an agreement with whatsapp.

**Comment by mvzeppelin** [comment]
I've talked with Whatsapp support team asking for unblock my number and I just want to share the official response aboult this issue: 

"##- WhatsApp Support -##

Hi,

Thanks for your message.

It appears that you are using an unauthorized application of WhatsApp. Unauthorized applications are not supported; they contain source code which WhatsApp can not guarantee as safe. Your private information is potentially being passed to 3rd parties. In addition, these may not function correctly and may cease to work at any given time.

Please uninstall your application and install an authorized version from:

www.whatsapp.com/download

After installing the authorized version, please reply to this email with a screenshot or an image of your screen from inside the app.

You can refer to your phone's user manual for detail steps of taking screenshots on your device.

Please note WhatsApp only works on smartphones, non-telephony devices including tablets are not supported.

Best Regards,

WhatsApp Support Team"


I have to agree that it's a matter of time to wa16 protocol being blocked as well.
:(

**Comment by Josue-T** [comment]
Hello, I have revived the same message this last night ;-)

But I don't know what I have to do, because I don't want to be blocked again

**Comment by softcode2** [comment]
here same problem 
all number was blocked

**Comment by thundergreen** [comment]
After sending Screenshots to wa they unlocked  my number again.now I need
to be sure the API works because otherwise they will totally block my
number

Thorsten Fröhlich
Am 19.04.2016 11:40 schrieb "softcode2" <notifications@github.com>:

**Comment by assegaf** [comment]
ok

**Comment by ksulanc** [comment]
In the past Whatsapp already announced that they don't want to make an API for developers so it should not be possible for third-party programms to use Whatsapp on any way.. the Yowsup Project was for my use case a nice way and good tool, but now I think the Whatsapp team is purging everything because of security reasons or whatever :(. Until the Whatsapp company is not publishing any API for developers there will be many problems like this and I think I am talking to every one here when I say that no one of us is having fun to update his scripts or the plugin everyday because of this problems like this. Sometimes it would realy nice when something is just working and you dont need to care about it every 2 days.
I loved the yowsup project realy much and will still watching here continuesly what is happening, but I think it is a good alternative way to use "Telegram" they also have a working API.

**Comment by thundergreen** [comment]
+1

I totally agree with you.I guess there's no working branch /repo atm and
will maybe never be. Or do the devs here think it will work again?

Would be nice having an overview for a working branch / repo . don't wanna
risk to get blocked to wa as I only have one number to register to.

If there's a working branch / fork or repo.please let us know .I really
appreciate the work you all did!
Thank you so much !!!!!!

Thorsten Fröhlich
Am 21.04.2016 08:49 schrieb "K. S." <notifications@github.com>:

**Comment by jegue** [comment]
Yeap already moved to Telegram! very good api 2.0; many many nice functions like the keyborad and
Inline mode. I love yowsup too! but until the the capistalist WA dont free us with a Official API! Telegram will grow!

**Comment by thundergreen** [comment]
Would like to know if someone here uses telegram spectrum2  as a Transport.
I don't get it installed / compiled on my Debian Jessie system ..any help
here?

Thorsten Fröhlich
Am 21.04.2016 12:45 schrieb "jegue" <notifications@github.com>:

**Comment by sandman6668** [comment]
Hey Guys,
there's no need to worry. I successfully registered my phone number YESTERDAY using yowsup and everything is working fine. I can send and receive messages and don't think I get blocked (fingers crossed :P).
I already used this number in the past with an older version of yowsup and cannot refer to the "blocking issue". So you can ask me if you have any questions.

Btw: I'm using the wa16 branch with a few adjustments.

**Comment by thundergreen** [comment]
@sandman6668

Does file transfer work for u? Picture send.and which modifications are u
using? Would give it I try if your yowsup still runs after one week ;) keep
us posted

Thorsten Fröhlich
Am 22.04.2016 13:43 schrieb "sandman6668" <notifications@github.com>:

**Comment by sandman6668** [comment]
Yeah, file transfer works great even with encription. :)
The only thing you have to modify is the MediaUploader. It has to be threaded in order to be indepentent of its main thread. This is necessary otherwise you get a ping timeout if it - for example your video file - takes too long to upload (and to be capable of sending multiple images/videos at once).

`thread.start_new_thread(mediaUploader.start())`

I'll keep you up-to-date :)

**Comment by thundergreen** [comment]
@sandman6668 which layer or file do i have exactly have to change how 
because i'm more like a noob here :P

and which fork / repo / branch exactly are u using? would like to give 
it a shot u know :)



Am 22.04.2016 um 15:05 schrieb sandman6668:

**Comment by softcode2** [comment]
@sandman6668

Still all going well with you with out blocking  ?

**Comment by thundergreen** [comment]
https://github.com/sandman6668

Still working?
Am 23.04.2016 21:00 schrieb "softcode2" <notifications@github.com>:

**Comment by sandman6668** [comment]
Everything was working yesterday ;)
I'm currently not at home so I'm unable to give you an update for today but I will tell you more tomorrow.

I also fixed the emoji issue.

I will answer your questions as soon as I have time. But as I already said: I'm using the "official" wa16 branch of tgalal, fixed the AndroidEnv (I think it's already fixed if you pull the latest version) and then you should be ready to go ;)

**Comment by GHswitt** [comment]
Registered yesterday a number, was working until now, approx. 24h. Now I get "Auth Error, reason not-authorized" and trying new registration is showing "INFO:yowsup.common.http.warequest:{"status":"fail","reason":"blocked"}". Only received/sent messages from one number, and not many (<100), with time between them.

**Comment by softcode2** [comment]
@Ghswitt
How you was registered that new numbers? 

Maybe you send more that 50 msg .
That's why you got blocked !

**Comment by GHswitt** [comment]
@softcode2 
It was registered with yowsup-cli. And yes, the number sent more than 50 msg. But not in one direction, it was always question (normal phone) -> answer (yowsup). And not really fast.
So why can one only send 50 msg?

**Comment by softcode2** [comment]
I think per day you have limit . I am not sure about this but it was like this .

So I think that's why you get banned .

**Comment by rami-dabain** [comment]
Looks like revistering with new version then using wa16bgets you blocked. 
I registered 2 numbers yesterday (i really dont want to buy a smartphone) didnt use them... this morning they were blocked. i only logged in and synced 5 contacts

**Comment by boppreh** [comment]
Worked for a day and a half. Sent 10 messages, received 7 (all to and from the same number and all typed manually), then today it got a "system-shutdown" XML message and registration is now blocked. The number was from a real phone.

@mvzeppelin
> It appears that you are using an unauthorized application of WhatsApp. Unauthorized applications are not supported; they contain source code which WhatsApp can not guarantee as safe. Your private information is potentially being passed to 3rd parties. In addition, these may not function correctly and may cease to work at any given time.

This makes my blood boil. They are violating net neutrality around here and getting exclusive deals with mobile network carriers, so there's no way to compete with WhatsApp anymore. Now they are kicking out alternative clients. The security remark is a red herring, I trust the open source Yowsup more than their closed client. And the whole paragraph, specially "may cease to work at any given time", implies it's not their action and that's just disingenuous.

**Comment by yniwly43** [comment]
I can confirm that the /develop branch (which is basically wa16) also causes banning. :cry:

**Comment by e0xify** [comment]
yea same here, got unblocked yesterday again, yowsup worked for like 10 minutes, then i got "auth error" and by new registering i get "blocked" again...

guess its like the 3rd time.

**Comment by tgalal** [comment]
I don't think yowsup causes banning. Rather the way you use yowsup is what causes the banning...

**Comment by e0xify** [comment]
guess using it the same way, others do. 
this happens anyway, just starting the yowsup-cli demos -y -c config.yowsup, recieving like 10 messages --- starting to get auth error, blocked.

**Comment by mcbastian** [comment]
@tgalal I'm afraid you're wrong here. Signed up with yowsup, sent one message, signed out. RPi was off till next day. Booted, tried to sign in, blocked.

**Comment by k-freeman** [comment]
i think "invalid signatures" are producing bans. These are messages, that are not ACKed. New messages are ACKed => inconsistency, that whatsapp might see...

**Comment by tgalal** [comment]
Well there must be a reason me and many others never get blocked

**Comment by k-freeman** [comment]
> Well there must be a reason me and many others never get blocked

are you using yowsup in groups? are you sending your yowsup client voice messages or are you receiving "invalid signature" errors? this happens often in groups :/

**Comment by mcbastian** [comment]
@tgalal It would be helpful to give a short description on how you use yowsup... How do you sign up? how do you enter contacts? how do you use yowsup?

**Comment by tgalal** [comment]
For the last 3 months, I've been working solely on integrating group enc, so yes :)

**Comment by k-freeman** [comment]
and you are not getting random invalid signature errors?
> [WARNING][2016-04-15 08:28:17,991] Invalid signature!
This is what I mean :) thx

**Comment by e0xify** [comment]
like k-freeman said, maybe its related to the groups, got 2 groups running with using yowsup, getting bunch of invalid signature sometimes... only recieving messages by groups by the way.

**Comment by mcbastian** [comment]
I haven't used groups and I get banned, even after just signing up and sending one message...

**Comment by mcbastian** [comment]
tried with 4 different numbers

**Comment by tgalal** [comment]
I've never seen that Invalid signature so far, I'll keep an eye on groups

**Comment by e0xify** [comment]
@tgalal if you want i would grant u access to my vps to check any logs you can find and see the errors by urself.

**Comment by k-freeman** [comment]
you can easily reproduce this:
create a group, solely with a yowsup client, spam some messages (like 10-30, happens rarely) and you will see some without delivery ACK to the client... i don't know if it is implemented in your master right now, but it happens also if you send a voice message to the client directly.

**Comment by tgalal** [comment]
@e0xify See, here is a clue. [vps](https://github.com/tgalal/yowsup/issues?utf8=✓&q=vps)

**Comment by e0xify** [comment]
worauf willst du hinaus?

**Comment by assegaf** [comment]
almost all vps ip is banned by whatsapp,
so you must run on your own local internet line like home isp

**Comment by e0xify** [comment]
aight then i will try if with my rpi today

**Comment by e0xify** [comment]
the issue with the periodic server ping to whatswapp servers is already implemented by default, isnt it?
strange issue is, that it worked before with the vps for a like 6 month.

**Comment by k-freeman** [comment]
i was blocked, using my own internet @ raspberry pi.

**Comment by tgalal** [comment]
@k-freeman develop branch

**Comment by k-freeman** [comment]
@tgalal  I supposed that means "yes" as well? :-D i'll try it out and see, if there are invalid signature messages :) thanks

**Comment by mcbastian** [comment]
I am running via a (fixed) DSL-IP from T-Online. It's our company network with ~50 employees using WhatsApp (official) on their smartphones without any issues. Only my RPi within the same network gets blocked after a certain amount of time. (3-6hrs after registration).

**Comment by GHswitt** [comment]
Is their some guideline how to use yowsup to not get banned? I really didn't set it up and spammed hundreds of numbers. I only sent messages between yowsup and one number (< 100).
If this means I was using it wrong, then there really should be an example of how it should (not) be used.
For example if the python side should add a random delay before sending an answer, so it does not get detected as an automated system.

**Comment by zieglersystems** [comment]
There must be a trick, as the German provider of whatsapp broadcasting (https://www.whatsbroadcast.com/en/) is using whatsapp for months and is not getting blocked. They send about 5 messages per day per number to a huge number of receivers...

**Comment by k-freeman** [comment]
@tgalal just tried it, cloned the develop branch and tried the echoclient. send a voice message in a newly created group =>
> ERROR:yowsup.layers.axolotl.layer_receive:Invalid signature!

**Comment by tgalal** [comment]
@k-freeman generally encrypted media are not supported yet

**Comment by tgalal** [comment]
@k-freeman  And I'm still not getting this error for some reason, even following the same steps you described. Ideally it should crash after decryption with something like "Unhandled", Invalid signature means it could not even decrypt. Please run echo client with ``--debug``` argument

**Comment by k-freeman** [comment]
@tgalal here you go:

`DEBUG:yowsup.layers.logger.layer:rx:
<message participant="49xxxxxxxxxx@s.whatsapp.net" notify="Kevin Freeman" t="1461752681" from="49xxxxxxxxxx-1461751619@g.us" type="media" id="B62CA1D8F3A14AC60E274FE246B74F" phash="1:qCFXpk0n">
<enc type="skmsg" mediatype="audio" v="2">
���j�w��\���KF���M���]?}C�������u�w��^���
Te������(���                                ���e�z-�v�� LT��-[]�aj�����9S/K�5F8�㜛�
             /�6!V���&@,=�l�)_'1*����@!��
                                             �n��L�,�:F���@�I0Qo�a�� �nF�mQ������~/������V*�v~��1iȤMvoe��)sw��Qr��Yi�MG(��K�+�R���,Q�8�6go_Ǣr�;�ܡ�ą�_եL�X_�Hy���JN�[�R�5�|j�۠x�|D-��%���Be��p���4߽>��g���V��N��;��N��v�N9�8�lg �Y�f40�-_~]�>�h�v|�fe�E�W����Hg�%>{ϕ�ݕ�-����OV��o��$�++���c�����E���jN[^����I�?=Q�E�wɘN���x��ovYy6��Vsá�n�7�U偎
HEX:3308f68ef16a10051aa00377a2c65c8df2bd4b469ab8b04db3bacd075d3f7d43ebecfdbea400e8de75f577eda05e15bae8af0c8a181afcd31b65af7a052df576dae6091c004c540eb413a92d5b5d1be861186aafb2e906a1b93953072f4b9535461c38b0e39c9bed0d1054658eb8fdfcaaf22882b0fb0b2fd736215614e98f079d2601402c3ddd6cc729165f27312aa213bdadd54021a0ee0be76eb8e34c118b2cdb053a4687f6ba4018c4144930516fd0619985e89f9408d96e46e06d51b7f2a9e70edf17f67e2fbac8defb84a01e562afd767ea4893169c8a44d17766f65e4d829737791875172a6cf5969e44d47285c0805fbb84bb52bdd52fbe29b2c5117f4388236676f5f16c7a2721782143be0dca1e9c485c0155fd5a54c0ed0585f9048790206b019d6d24a4e17109a5ba352a235c57c6ae4dba00378fc7c442dd2f12583ece34265addf7090bbe134dfbd3e0ee4d46797888556ecd64eddd13bb5ed4ee7b01776b84e391dd438fb6c6720d2599b6634309c2d5f7e5d873efa68d2767cac66658c45d657decb01bee24867f2253e7bcf95f9dd95ac2d849fba904f5697b96f849f24f3062b022b1bc4f1b963f71bbfb9b9bc458ab7b26a4e5b5e81151fd0fecb49f53f3d51ba45c077c9984e13fdf2c67892ff6f76597936cdef5673c3a1ae6ee937b955e5818e
</enc>
</message>

ERROR:yowsup.layers.axolotl.layer_receive:Invalid signature!
DEBUG:yowsup.layers.logger.layer:tx:
<receipt to="49xxxxxxxxxx-1461751619@g.us" participant="49xxxxxxxxxx@s.whatsapp.net" id="B62CA1D8F3A14AC60E274FE246B74F" type="retry">
<retry count="1" t="1461752681" id="B62CA1D8F3A14AC60E274FE246B74F" v="1">
</retry>
<registration>
%�X
HEX:1925ad58
</registration>
</receipt>

DEBUG:yowsup.layers.logger.layer:rx:
<ack type="retry" participant="49xxxxxxxxxx@s.whatsapp.net" from="49xxxxxxxxxx-1461751619@g.us" class="receipt" id="B62CA1D8F3A14AC60E274FE246B74F">
</ack>

DEBUG:yowsup.layers.logger.layer:rx:
<message participant="49xxxxxxxxxx@s.whatsapp.net" notify="Kevin Freeman" from="49xxxxxxxxxx-1461751619@g.us" type="media" id="B62CA1D8F3A14AC60E274FE246B74F" t="1461752682">
<enc count="1" type="msg" mediatype="audio" v="2">
3
! �hF�#]�U=������w�_B�+��gL���{"�2�$9^}^I�q�ي������D�A��� ���`����>���)���Z|��c�=�e�Gi�cN�uY�'HH/"�~w����:
>��˔���Igɬ#ZCە��?����VG�9Z��6t��L���F_�^�.���
F�%�(��Na���d7� \
                      ��O�[�
                             ����SM�I�ߣ�7!c� �`��ρg��e�X�Iw���9���������0�5�Y1� o�t4��a�e�=+��'���*=fb�A���(O�[*����c����}�0��)�'�5\^�x
�RA�~m��o�����+O����q}�   a(�� ���U�dk?qK|�Q���Uȝ�?D䬉���   |������-��9
����rz������w�
               4�@�S?�,��.m!����3��=]��,Y�^(��>YJ��
                                                      ��J���丩�R��V��A���~����CꆾI�n�»[)��Au/�Gt9�\!�
HEX:330a210520f0684692235dcd553daf10d4f4a0abb077ba5f0e42df2b9ee3674caea5e37b1001180122e00332e824395e7d5e49b2718819d98ac4108ca0b9f08a44ee41f7f4dd20e60512e28a60e3e611cbd73ebdc1cb29b795c25a7cb09663c43d00d765f84769e3634eb17559f02748482f22b17e77aefae4873a0a3eec84cb94d7f6e6490367c9ac235a43db95d203de3fac9e84965647b3395ae4db367481049c4c93e6c5465fcf5eee2ebcfab40a461d91258b28f38f4e6106f11fb8a9641c37179c205c0be5c44fa21b5b990bdfe7ee00dc534dc349cbdfa3913721638220de60a3a2cf8167e1830e1565e558e14977e3c5c839e099c5d5cee5f9dffd30d835845931cb206f07a67434f717e261c665cd7f3d2b9cce27a4f99b2a3d660462a3411be813f7b0284f1cc7ef93a85b2a8db096c563cee4cef17dc230ba0fa629d027b0355c0e5e17f7780b39e7d91acb2b4ffad1e891717dfb09e0b1a3612882ad162083c0af55f7646b3f714b7cff51d619ca06a655c89d00a73f44e4ac899b8b98097c8cdad2e191f02dcaed390dbf1b5241977e6df48c6fdcfb0a198aa6bae2727ac3febab589eb77e90b34dc40ec531b3fdb2cf910069d2e1c086d21ddf7c6e83382c60f3d5db6882c59aa5e28e2893e594ac1b60bc08e4aa9aa06a7e4b8a9d6527f8f198556bf1214a341e1f9f17e87f6929043ea86be071049ce6e1389c2bb5b29ecd27f41752fc3477439f15c21d5
</enc>
</message>`

**Comment by tgalal** [comment]
@k-freeman 
So if you look at the sequence: 
- you receive audio message
- invalid signature
- yowsup sends a retry
- you receive audio message
- do you get invalid signature again ?

**Comment by k-freeman** [comment]
@tgalal 
no, but the client never receives the correct message (e.g. text message) and therefore, never sends an ACK. the message remains unACKed, even though new messages come..

**Comment by tgalal** [comment]
@k-freeman what client ? you are receiving the voice message on yowsup. Do you mean if the messaged is echoed ?

**Comment by k-freeman** [comment]
@tgalal voice message has been just an example to get the invalid signature error direclty. when using the cli demo, i get the same error for (some, only a few) text messages. (invalid signature). these messages are never seen at the client in cleartext.

**Comment by e0xify** [comment]
allright, i installed all dependencies on the rpi 3 now, python 2,7 yowsup axotlol and stuff.
requested another unblock at whatsapp now. i guess its the 5th.

will try it then again. i guess it will get blocked again.

**Comment by ashishnetworks** [comment]
@k-freeman I was getting warning messages like ERROR:yowsup.layers.axolotl.layer_receive:Invalid signature! but I Ignored that as my program was working fine and then disaster happened my number got blocked !! and registered another number but again same few massages and blast again blocked. So you meant that message was harbinger of block right? 

How do I prevent my program from producing that working ?

**Comment by k-freeman** [comment]
@ashishnetworks i think whatsapp is able to see the incorrect order of acknowledgements. so if a message with invalid signature is received, it has to be acknowledged. in the develop version it is working after a retry of the message (see above) but it doesnt acknowledge the message as far as i can see.

**Comment by tgalal** [comment]
@k-freeman good point

**Comment by tgalal** [comment]
But, we are in fact sending receipts (retry receipt is how you ack a messages). Acks are for when **we receive** a receipt, which are also sent, unless you found a missing ack somewhere ?

**Comment by k-freeman** [comment]
@tgalal see at https://github.com/tgalal/yowsup/issues/1516#issuecomment-215042504 again. this is the full log of a invalid signature message. it is received, retry is sent, ack is sent type: retry

no ack "read" is sent, ever.
it should be <ack ... type="read"> after receiving the correct message (after failure of signature). am i right?

**Comment by tgalal** [comment]
@k-freeman That's true, after receiving the correct message, a receipt should have been sent.  Were you using one of the demos? This should  be handled by the interface layer. For example in [here in cli client](https://github.com/tgalal/yowsup/blob/master/yowsup/demos/cli/layer.py#L525), is

**Comment by k-freeman** [comment]
@tgalal yes, I tested the demos. mostly testing on demos -y (cli)
in jlguardis version, that line of code is never executed. but I can test it for your version as well. second.

**Comment by tgalal** [comment]
Actually as it's an audio message, it would never execute. It doesn't go all the way to interface layer as encrypted media are not yet supported.. If you use my version. it should crash with something like "Unhandled value" message.  Perhaps jlguardis's version is tweaked to not crash in this case, should also send the receipt anyway though.

**Comment by ashishnetworks** [comment]
This is the order in which I send my ack, do you suspect something fishy here ?

```
`class VirtualAssistant(YowInterfaceLayer):
    @ProtocolEntityCallback("message") #once message is received it starts executing following code
    def onMessage(self, messageProtocolEntity):

        message = messageProtocolEntity.getBody()
        sender = messageProtocolEntity.getFrom()  # full id with whatsapp server domain
        sender_mo_no = sender[:12]  # only mobile number
        msg_received = OutgoingReceiptProtocolEntity(messageProtocolEntity.getId(), sender, 'read',
                                                messageProtocolEntity.getParticipant())  # read,and delivered receipt is constructed here
        self.logMSG(message,sender_mo_no)  # message is logged here
        self.toLower(msg_received)  # read,and delivered receipt is sent here, OTHERWISE WE KEEP RECEIVING SAME MESG OVER AND OVER


        im_typing = OutgoingChatstateProtocolEntity(ChatstateProtocolEntity.STATE_TYPING, sender) #Here typing status is constructed
        self.toLower(im_typing) #typing status is sent here'
```

**following code at the ending of class
**
```
@ProtocolEntityCallback("receipt") #This+Following code sends "status received message"
    def onReceipt(self, entity):
        ack = OutgoingAckProtocolEntity(entity.getId(), "receipt", entity.getType(), entity.getFrom())
        self.toLower(ack)
```

**Comment by k-freeman** [comment]
@tgalal yeah, it doesnt go there. just tested it. however, i think that this might be the reason for blocks, but could be that it is not. i think i saw a wrong ack id when using jlguardis version, could be, that this is the reason for a block...
however, you say you didnt get blocked, even though some messages are not acked (and only acked upon restart of the client?)
thank you for your time :)

**Comment by tgalal** [comment]
In my case, I haven't been receiving any media messages, only text, which means messages were always acked.. Might be that in your case you sent new messages without acking received ones (in case jlgaurdi version doesn't crash on receiving media, and doesn't also ack them)

**Comment by k-freeman** [comment]
@tgalal yeah, i send / received new messages without acking in between. so is there a possibility to enforce ack'ing the media message somewhere? i think this would be very useful until media message are able to be parsed again :)

**Comment by e0xify** [comment]
guess thats the issue then. 
pursued the conversation, getting banned the same way, after sending / recieving media messages in groups.

**Comment by jlguardi** [comment]
@tgalal, @k-freeman my master can receive and read properly encrypted media (send is still failing) so you can ack them properly.

**Comment by softcode2** [comment]
HI guys 
my VPS IP blocked with number  .
using chat-api 

its general problem

**Comment by yniwly43** [comment]
I can tell you I got banned several (like 8) different times, using only REAL numbers used and registered with real devices first, using IPs ranging from my own home to different VPNs, never having sent even ONE message.

I just connect, set the presence to available, subscribe to some numbers for presence updates and wait for incoming messages (receiving, acking, reading) from those numbers, and that's it. Nothing more nothing less.

What would say that I do that could cause banning?

**Comment by yniwly43** [comment]
Also, leaving a number connected (login, presence available, that's it.. responding to pings, no messages or anything else) using branch develop I remained connected for 3 days with no issue.

Could it be the subscribes? The incoming messages?

**Comment by schoerg** [comment]
How can one be sure if a number is blocked?

```
[offline]:/login 436648561069 /pw=
DEBUG:yowsup.layers.network.layer:Connecting to e2.whatsapp.net:443
DEBUG:yowsup.layers.logger.layer:tx:
<stream:features>
<readreceipts>
</readreceipts>
<groups_v2>
</groups_v2>
<privacy>
</privacy>
<presence>
</presence>
</stream:features>

DEBUG:yowsup.layers.logger.layer:tx:
<auth passive="true" user="436648561069" mechanism="WAUTH-2">
</auth>

DEBUG:yowsup.layers.logger.layer:rx:
<stream:features>
</stream:features>

DEBUG:yowsup.layers.logger.layer:rx:
<challenge>
▒▒▒▒▒婯▒K¨F▒}p▒▒
HEX:e49cdd1fdcfcaae5a9afa24bc2a846e57d70d3fa
</challenge>

DEBUG:yowsup.layers.logger.layer:tx:
<response xmlns="urn:ietf:params:xml:ns:xmpp-sasl">
▒@▒
   ,▒▒B▒P▒▒P▒▒Zi▒x▒0▒▒k▒S.▒(▒hy▒ڦ▒▒w=▒ߌ[X▒(▒▒߰E$▒▒Ng▒
▒▒ywX▒y
HEX:a440970b2ccbcb1708de42bc50f7f050f6e45a69a678a7301cd3c96b84530f2ecd1428ba6879d1daa6a71cb2773d9cdf8c5befa19b58aa1bde288cafdfb04524f7d44e67c00ae5fb797758e50579
</response>

DEBUG:yowsup.layers.logger.layer:rx:
<failure>
<not-authorized>
</not-authorized>
</failure>

general: Login Failed, reason: not-authorized
DEBUG:yowsup.layers.network.layer:Disconnected, reason: Authentication Failure
Auth Error, reason not-authorized
```

However, if I use the same account with the same PW on a real phone with a real WA client it still works perfectly.

**Comment by k-freeman** [comment]
try registration, it will say "blocked" if it is blocked

**Comment by schoerg** [comment]
Thanks, it's not blocked and sent a SMS.

**Comment by k-freeman** [comment]
i dont know, working fine for me. maybe pull again / recheck yowsup/env

**Comment by k-freeman** [comment]
Okay, just found out, that adding someone to a group will give you an instant block. i tried to add someone, didnt work as i wasnt admin, gave yowsup admin rights, instant blocked. latest develop branch

**Comment by yniwly43** [comment]
ugh

**Comment by tgalal** [comment]
@k-freeman Did u sync the numbers u added to the group with yowsup ?

**Comment by k-freeman** [comment]
No, nothing. Just added the client to the group, wrote some messages, added
a Person via client.

**Comment by tgalal** [comment]
@k-freeman Try sync next time. Because on a phone using whatsapp, u can never add a number to a group unless it was under 'favorites', which means it was synced before.

**Comment by k-freeman** [comment]
Okay, i think i'll try that later. Maybe there should be an auto sync /
check when trying to add people in order to safe people from bans.

**Comment by e0xify** [comment]
@k-freeman how do you manage to get unbanned so fast? need to wait like at least 24h

**Comment by rami-dabain** [comment]
@tgalal What version donyounuse to register? because i got blocked without even using anything. Here are my tries:

1- Register, login, logout
2- register, login, sync 10 contacts
3- register, login, presence to available, sync contact send one message and receive one message (acks as well) i even set the "typing" while actually typing...

all registrations and logins were done from a fresh VPS using German sim card from vps in germany, russian simcard from vps in russia, us sim.card from a vps in the u.s. all were real sim cards

next day, all are blocked
i have only my main sim card which was registered a year ago. looks like if you register with wa16, they run a cronjob to block all who registered using it once a day. did you try registering a new simcard, if yes, and it worked after 24 hours, how didnyou register?
I guess we should do reegistrations using the latest version to avoid that, and really wee need to wireshark or unobfuscate their latest decompiled version as they might require some kind of a message through tcp to confirm registration. and all those whondid not confirm within 24 hours gets blocked? it is a huge probability

**Comment by k-freeman** [comment]
@e0xify I am not getting unbanned, I use different numbers.

@rami-dabain i used @jlguardi's master branch and/or develop branch of @tgalal to register. both worked fine for more than 2 days: i logged in, presence available and sent messages and received messages (over single chat) and kept an eye on ACKing in the right order. When receiving an invalid signature (as written above) I am currently restarting in order to ack this message (not a good way, but i think it'll work for my test scenario). my client is in a group again already, lets see how long this works ;)

**Comment by mcbastian** [comment]
I'll sum up my experiences:
- the IP is not (always) the issue as my RPi using @tgalal s wa16 is sharing the same network/public IP as dozens of other clients using the official App.
- groups might not be the issue as I didn't use them, in any way and always got blocked
- I get blocked (3-9hrs, ususally in the evening (CEST)) for just registering and sending one message....
and what I hear from other forums, (e.g. FHEM community) I'm not alone with this problem.

**Comment by ashishnetworks** [comment]
Now it has become severe issue. No use of anything. Got frustrated ! Nothing working. Getting banned every single time. We must do something about it and give this issue higher priority, otherwise there is no use of any feature if we keep getting banned.

**Comment by e0xify** [comment]
@tgalal took ur latest develop edition, if i get a message now, its spamming:

_layer_send.py_ [INFO][2016-04-28 14:50:18,370] Got retry to for message 1461847817-4, and Axolotl layer has the message

and after delay i get the message.

**Comment by tgalal** [comment]
@e0xify  That's normal

**Comment by e0xify** [comment]
@tgalal - is that normal as well?

[WARNING][2016-04-28 14:53:01,471] InvalidMessage or KeyId for ---- ?

**Comment by jlguardi** [comment]
I've a banned number registered 2 years ago and online since registration. I've no incoming/outcoming messages for 6 months. Just pings, presence subcribed to just one number and some reboots.

I've seen that official wa client sends Ids with format:
```
<notification type="picture" t="146181xxxx" from="xxxxxx@s.whatsapp.net" id="267961xxxx" notify="xxxxxxxx">
```
And messages has id with format id="8C3B1778D96577A4DCC42F434E71DE"

Probably this is the banning cause.

**Comment by tgalal** [comment]
@e0xify yes this is normal

**Comment by rami-dabain** [comment]
@k-freeman but i get banned without even doing anything, just register and leave it for one days before login and BAM, blocked. is that normal?

**Comment by e0xify** [comment]
running for me fine until now, running on rpi since 2 hours, registered new, recieved bunch of old messages, and now everything looks fine.

**Comment by k-freeman** [comment]
@rami-dabain if you're not doing anything it can only be two things: your IP/number or the environment variables (used for registration). don't know, i used develop branch and received + sending many messages over rpi3 since yesterday. seems to work until now for me. no files were changed except layer.py (nothing yowsup relevant though) (in CLI demo)

**Comment by rami-dabain** [comment]
Will give it a try today as well. (check my previous message about what did i do, from 3 different IPs)

**Comment by yniwly43** [comment]
Today I did a simple test.
Registered WhatsApp on an old phone for 3 numbers with 3 real SIM cards. Sent a message to my real account.
Using yowsup-cli: re-registered them using Yowsup (with correct, real MCC, MNC, etc) using Android env, latest develop branch commit. Put each one on a different IP. All from VPNs in the US.
Connected them all. Sync'd 30 contacts in one, 60 in another, 110 in the third. Set presence available. Presence subscribed to all of them. And left it running.

Almost exactly 24 hours after the registration, (~1400 pings done in total, one every 60 seconds), all were blocked.

This is the most minimal of messenger behaviors. 

Just like project yappari, which is a Maemo-based Whatsapp client that died due to the changes in the protocol.

Interestingly, I also created a single account in exactly the same way, almost 6 days ago. I then logged in and set my presence to available, and that's THAT. This number is still running, not blocked.

Based on all the tests I've made:

1. It doesn't seem to be the IP. (Maybe registrations are affected by it, but I doubt it.)
2. It isn't a specific clear cut behavior (we've seen many examples of naive, simple, "by the book"-flows that got blocked in the past weeks.)
3. It doesn't have anything to do with the ETE encryption, group messaging, etc not being implemented properly. We've seen lots of blocks being done to people who haven't sent a single message or had anything to do with groups.
4. I've looked at all incoming/outgoing messages in several numbers, and I've seen nothing out of the ordinary.

My conclusion is that the following options are the most likely:

1. The Android env in the /develop branch is flagged, somehow, and that might cause the banning.
2. There is a missing step in the registration or login, or perhaps some protocol message we fail to send out, that makes the session invalid and cause the accounts to be banned.

What do you guys think?

**Comment by mgp25** [comment]
1. IP is related
2. Flow matters
3. Is not related to e2e
4. Messages doesnt matter, but its very important syncs in both ways and timing.

1. Not flagged, but maybe people coding wrong flows
2. Registration is fine and login too
3. Although WA 1.6 is not the newest protocol, it will be available until the end of the year, and it works perfectly. IDs are okay at least if its the same as chat api

What causes a ban or flag is:

- Sending without syncing
- Sending a lot of messages to people that doesnt have added you
- Multiple messages in a row
- Bad flow behaviour
- Multiple registrations under the same IP causing IP flag
- Loginning in with lots of number under the same IP
...
Do i really need to continue?


My recommendations:

- **Use it for personal usage only**
- Good timing and flow

**Comment by assegaf** [comment]
is it possible that 2.12.556 is WA20 protocol in registration but we use WA16 Protocol in Auth ?

**Comment by rami-dabain** [comment]
@mgp25 
Prevet!
I registered ONE number from a fresh IP everytime, did sync before anything, connected only one number per IP, no messages sent/received (new sim cards, no one knows about themsim cards have been added to my phone, and synced with real whatsapp, and i synced only my number to them via yowsup) used wa16 to register and login on one number, wa16 to register and android to login on another number, all got blocked

**Comment by rami-dabain** [comment]
and i used the 2 sims i had before for personal use only, one for work (desktop client) the other is private (i don't want whatsapp on my phone, took a look into their code, man it copies everything to facebook)

**Comment by e0xify** [comment]
still working for me since 2 days

**Comment by k-freeman** [comment]
Three days or four days for me. Inviting, promoting, demoting, doing
everything except media messages.

**Comment by e0xify** [comment]
doing media messages as well, no probs yet

**Comment by k-freeman** [comment]
maybe the ban wave stopped? anyone got blocked in the past 3 days?

**Comment by softcode2** [comment]
Guys ..
Any way to unblocked my VPS IP. 
It's blocked by whatsapp before 5 days !

**Comment by dgua** [comment]
Just updated Yowsup today because previous version stopped working about two weeks ago. Went through the usual registration, but this did it did not work: I was blocked!

There's something different now: It seems Yowsup does not work anymore. 

This is most upsetting. Any credible plan to make it work again?

**Comment by yniwly43** [comment]
@mgp25, I am fairly certain any serious for-profit company using Whatsapp has better solutions than Yowsup. Probably they already reverse engineered the protocol.. most of the users here are simple for-fun or for-home project users. I am the latter. (I moved to Telegram, but it's not as popular nor as fun.)

As for your remarks about my tests:

> IP is related
> Multiple registrations under the same IP causing IP flag

- I used different IPs for the tests I did, luckily my VPN provider has like 1000 different exit nodes so it is easy.

> Loginning in with lots of number under the same IP ... Do i really need to continue?

- I never used more than 1 account per home server, so unless you're counting real WhatsApp accounts (on real devices) on the same network (and there were none during the test I made), I don't see how this can be a problem

> Is not related to e2e
> Sending without syncing
> Sending a lot of messages to people that doesnt have added you
> Multiple messages in a row

- I didn't send ANY messages and don't plan on sending any.

> Messages doesnt matter, but its very important syncs in both ways and timing.

- What if I move mobile provider and get a new telephone number, with the old phone? I sync ALL my contacts, but none of them have my number (yet). I did that twice during my life, and I never got blocked. I doubt that matters at all..

> Flow matters
> Bad flow behaviour
> Good timing and flow

- I showed you the flow, similar to a :
Login (first login)
Sync 30 contacts
Set presence available
Presence subscribed to all 30 contacts
Left it running.
Why would such a simple flow cause blocking?

> My recommendations:
> 
> Use it for personal usage only

- I do. I don't even send messages, it is all incoming messages to control a "smart home".

---

There's a serious problem with Yowsup (and Chat-API, and Yappari, and WhatsapiNET etc), and every day you see more and more reports of the same issue - blocking. 

The Chat-API guys have the "solution", but they're 'witholding' it to kill all spammers:
https://github.com/mgp25/Chat-API/issues/1840
Maybe that's your plan too, I dunno. If so, I sure hope it's working. 
I guess personal users can wait and not use Yowsup for a few months.. or maybe the API can just die.

Anyways, I am personally tapped out after many long hours of trying to understand this issue and help the other users of the API, so I guess this is enough from me.

**Comment by schoerg** [comment]
@yniwly43 I think profit-companies have more official methods to use WA.

**Comment by mgp25** [comment]
@yniwly43 man you dont need 30 contacts to control a `smart home` come on, i do that with a single number.

Profit companies use a very depurated version of Chat API, i know this because i receive tons of mails daily related to this. And if you have moved to telegram, stay in telegram as you only need to control a "smart home" am i right?

Im sorry to be rude, but i always say, if you people don't understand what they are doing they wont success, and if all people keep doing the same thing they wont get different results.

I've been using a number for years without any issues, and i can say 100% sure its people issue and not the API, believe it or not, i develope this Chat API for almost 3 years now, so i know what im talking about.

API wont die, we are waiting to kick out spammers, so when its stricly necessary i will release WA 2.0

Regards

**Comment by yniwly43** [comment]
@mgp25 I synch'd the 30/60/110 leads as one of several tests, in a set of tests, to try and see what causes a block. I tried to pinpoint the reason, and with 1$ cost a test, it was something I was willing to pay to try and help all the disgruntled users of the API. In fact, speaking of the code of the actual system I used - I didn't sync *at all* in the last year or so. Not even once. That is a broken flow, obviously, but it didn't matter. It worked. And I sent exactly *0* outgoing messages.

Also, FYI, I never used Chat API, I used Yowsup. I cared about a working Chat API as a manner of studying it to fix Yowsup.

If a for-profit company depends on Chat API, then it is managed by numbnuts. Why? Because obviously these APIs are unstable. I've had various disconnections and problems with old tokens during the last year or so. It makes sense, this is a *personal*, *for enthusiast* codebase. Basing a company on it is like basing it on a house of cards.

That being said, the attitude of "stupid people don't know what they're doing" is not productive. There has been a surge in "blocked" reports and new such issues and tickets, and if I'd care enough I'd show you a graph of the count of such issue reports each month in 2016, and how there'd been a huge spike in April. I am sure May will be the same. There are many open source projects that are basically dying:

Check out messenger_bot:
https://pypi.python.org/pypi/messenger_bot/0.5.0.1
Check out Yappari:
http://talk.maemo.org/showthread.php?t=94465

And there are other examples. These all *worked*, they are personal, for home usage, and are definitely not "for profit". Yet they're broken. And these projects WILL die.

You keep talking about a correct flow, yet accounts get banned following the exact flow defined in Chat API's Wiki. Did you try to replicate your flow with a new number, or maybe you're basing your results on your 4-5 year old number? Did you consider that maybe the issue could be caused by a distinction between new numbers and old numbers?

I tried to fix Yowsup because I care about open source and I found this project very useful, many many months, and I wanted to pitch in and give back. It was a challenge and a personal quest. But, as you said, since I've moved to Telegram, I don't care as much. Thing is, my friends and family had to install Telegram, which they didn't have before.. that was a pain. But, that is already done, so whatever. 

Anyways, as I said, I'm not gonna bother with this much more. Good luck, and let's hope things get better.

**Comment by dgua** [comment]
It is now apparent from this discussion that Yowsup isn't viable anymore. Telegram looks a lot better, so I am trying that. It was good until it lasted. And f**k WhatsApp.

**Comment by GHswitt** [comment]
I tried again and only used echo/send demo from yowsup-cli, got blocked again after one day. I followed this issue and the summary I get from it is: If you get blocked, you are doing it wrong. Ok, if it's that difficult to use, I won't try again. I hope it works for those that really need it. Good luck.

**Comment by assegaf** [comment]
hmm rather focus on block issue, that happen now,  no matter how, official client also blocking if u send too many message to contact that are new, 

lets focus on other thing that more important like wa16, end2end encryption to all kind of message, media, etc, new feature upload doc, pdf,  .. and wa20 protocol that are new, lets find some data to develop it together about this noise protocol, new kind of ssl protocol they said ..

**Comment by chriso0710** [comment]
@GHswitt That means that even the current yowsup cli demo is doing something wrong...

**Comment by ashishnetworks** [comment]
can anybody giv me any link where I can learn Telegram API as fast as possible. I am in strong need of immediate migrating my projects from Yowsup to Telegram API

**Comment by tomerb15** [comment]
I strongly recommend @tgalal and @mgp25 to log at the banned use case inside #1571.
It is very detailed and i couldn't spot anything that is "Ban Human Behavior"

Both Yowsup and Chat-Api are great projects that many people want to use and help improve.
Therefore I think that some kind of a test-case repository should be established in order for all the users to share banned experience and MORE IMPORTANT, share NON-BANNED experiences.

**Comment by dgua** [comment]
ashishnetworks, just google it. 

So far Telegram works so much better.

**Comment by ronromer** [comment]
guys what is the more active telegram project? I'm in this one
https://github.com/vysheng/tg but so many issue open without response.

Regards,

**Comment by ashishnetworks** [comment]
@ronromer This one is easiest and also officially recommended API by teligram team, https://github.com/nickoala/telepot . It has plethora of services already available which you can eevr dream of having in any API.

**Comment by k-freeman** [comment]
> Both Yowsup and Chat-Api are great projects that many people want to use and help improve.
Therefore I think that some kind of a test-case repository should be established in order for all the users to share banned experience and MORE IMPORTANT, share NON-BANNED experiences.

I haven't received a ban since 10 days now. So, this is what I do:

Login, presence available, sometimes (every 1-2 days) sync 7 contacts, in multiple groups, sending multiple messages (no media messages though), kicking people from group and invite some, changed profile picture, using my own home IP.
I am not setting my state to "typing" when i want to send a message. I am not contacting people that are not in my synced list. That's it, more or less.
Ah, and: using develop branch of tgalal.

**Comment by decklord** [comment]
@k-freeman but are you able to register new numbers now? As many said before, we are not able to register new numbers, all are getting banned right away. Using different IP's per number. Temporary unavailable first, then blocked.

This issue started to happen a couple weeks after whatsapp enabled end-to-end encryption. I don't use the API to send spam.

**Comment by k-freeman** [comment]
@decklord I just tested it for you and it worked (1minute ago). no problems, not banned. or am I supposed to do something after i registered?
greetings.

**Comment by rami-dabain** [comment]
@k-freeman yup, wait for 24 hours or so after using it

**Comment by k-freeman** [comment]
@rami-dabain ok will have a look at it tomorrow then and report back

**Comment by rami-dabain** [comment]
I have personally switched to telegram, even my personal number, better API, more features, they don't block you!!! and you can even use it in browser while ur phone is offline and battery dead!, not talking about the possibility of using it on a tablet (which wasn't the case with whatsapp)

**Comment by thundergreen** [comment]
ATE u using telegram Transport? Like with spectrum2?

Thorsten Fröhlich
Am 12.05.2016 13:41 schrieb "Rami" <notifications@github.com>:

**Comment by rami-dabain** [comment]
@thundergreen just the basic Telegram API - not Bot API or anything else https://core.telegram.org/api#telegram-api

**Comment by k-freeman** [comment]
btw got blocked real quick, dunno why. only registered the acc.

**Comment by rami-dabain** [comment]
I feel like most of people will switch to telegram as a lot of businesses switched to telegram and asked their clients to contact them using telegram.
whatsapp is making a mistake

**Comment by schoerg** [comment]
I doubt that.

**Comment by chriso0710** [comment]
I am getting annoyed by these constant blocking issues. Using WA with "unofficial clients" is becoming unstable and unreliable. yowsup does not seem to have a bright future. Looks like we are fighting WAs "windmills" here. chat-api has the same problems, so no success switching to another 3rd party WA lib.
  
I am thinking of abandoning yowsup and switching to Telegram for my projects. Telegrams APIs are open and look very good. And they have a fantastic bot API too. There are already lots of libraries available.
Of course they won't block you, they encourage you to use the API :-)

Telegram still needs more users and a larger community IMHO.
So I would ask everyone to at least take a look at Telegram and maybe support it in your projects! 

Just my two cents...

**Comment by okmiim** [comment]
hi,
my number got blocked after using an advanced form of the echo demo.
I was only about 1 hour each day in the last week online.
And now I got banned between yesterday and today without being online.

**Comment by simonejan** [comment]
Telegram is way better and much more reliable, they use open source and open for the developers community.They respect the rules of MIT.  We should all abandon those USSR guys still acting like KGB. And help to promote Telegram all over the world. Let the WhatsApp guys learn a lesson.

**Comment by rami-dabain** [comment]
Cant agree more

**Comment by thundergreen** [comment]
I agree

Thorsten Fröhlich
Am 16.05.2016 18:30 schrieb "Rami" <notifications@github.com>:

> Cant agree more

**Comment by biji** [comment]
Banned again 😭
I was using wa16 branch

**Comment by thundergreen** [comment]
I think yowsup now is useless as there's no safe way to use it anymore

Thorsten Fröhlich
Am 19.05.2016 14:06 schrieb "biji" <notifications@github.com>:

**Comment by javadevs2016** [comment]
This is sad, Yowsup/Chat-API are awesome projects. I don't think this is a problem with incorrect use of the API, as some of the guys in charge suggested. I got blocked after only using the registration tool. Not a single message sent or received. It seems to me that Whatsapp is actively hunting down API users. It seems odd to me, seeing how they process (allegedly) billions of messages, how on Earth are they identifying API users? We are missing something.

**Comment by biji** [comment]
Another number blocked, using develop branch
Very low traffic, after 1 month
I got stream error before blocked

**Comment by Gab0** [comment]
@biji Did you use to sync your contacts?

**Comment by biji** [comment]
Yes only once

**Comment by pcortizona** [comment]
Can we all place the lines in debug mode to capture timestamps. This way we can find correlation between the bans and a possible polling of the accounts is done with a  message  that the api can't decode

**Comment by chriso0710** [comment]
Bye bye, Yowsup. I am leaving. It really was a bumpy ride.

Yowsup worked well for a year, but now it is practically useless IMHO. The amount of time I spent to try to keep my yowsup client (just for receiving messages) running is becoming ridiculous.
 
I don't understand Whatsapp's policy of blocking 3rd party clients, when they open up the Facebook Messenger at the same time with a new bot API (https://developers.facebook.com/blog/post/2016/04/12/bots-for-messenger/). 
Maybe Whatsapp will someday publish a business API, which makes it easier for us to connect to it. See https://blog.whatsapp.com/615/Making-WhatsApp-free-and-more-useful?l=en&set=yes

In the last couple of days I made a Telegram bot with Telegrams great API, Ruby and https://github.com/atipugin/telegram-bot-ruby. That was easy, fun and the bot is working great. This was simply no comparison to my Whatsapp-yowsup experience.

So farewell, yowsup, until we meet again ;-)

**Comment by thundergreen** [comment]
Hi Christian

I would highly appreciate if u could help me to setup this telegram a pi
bot. Are u using this as xmpp transport?

You can reach me under xmpp:thorsten@emevth.no-ip.biz

Greetz

Thorsten Fröhlich
Am 30.06.2016 09:49 schrieb "Christian Ott" <notifications@github.com>:

**Comment by k-freeman** [comment]
@thundergreen , look at my github repos. Abb1t is a modular telegram bot. just clone it :) it is easy to use and the modular structure is self explanatory i think.

greetings

**Comment by thundergreen** [comment]
Also with xmpp?  Sorry I'm new to this

Thorsten Fröhlich
Am 30.06.2016 12:43 schrieb "Kevin Freeman" <notifications@github.com>:

**Comment by k-freeman** [comment]
@thundergreen no, it is written in python, just as yowsup

**Comment by thundergreen** [comment]
So it's hard to use it as a xmpp transport right?

Thorsten Fröhlich
Am 30.06.2016 13:06 schrieb "Kevin Freeman" <notifications@github.com>:

**Comment by k-freeman** [comment]
if you used that in yowsup, you can easily port it to telegram, as my client is using python as well as yowsup does. there are however packages for xmpp support in python, which you could use as well...

**Comment by softcode2** [comment]
Most problem that whatsapp famous
So little who use telegram
Also in my country that telegram blocked
بتاريخ ٣٠‏/٠٦‏/٢٠١٦ ١٠:٠٧ م، كتب "Kevin Freeman" <notifications@github.com>:

**Comment by sowerkoku** [comment]
[see here](https://github.com/tgalal/yowsup/issues/1661#issuecomment-228287130)

**Comment by MrGlue** [comment]
You will be blocked if you just register with yowsup and do nothing,  right? Maybe it is possible to extract the pw from the official app and use it to login with yowsup?

**Comment by schoerg** [comment]
@MrGlue Depends I guess, my registration worked fine.
You can't get the PW from the official app anymore.

**Comment by sowerkoku** [comment]
[this](https://github.com/mgp25/Chat-API/wiki/Extracting-password-from-device) no longer works ?

**Comment by MrGlue** [comment]
@schoerg are you sure? I used @sowerkoku 's linked apk. The app threw the error that the file pw did not exist. I looked into the directory and found a file called key. Duplicated it to pw and the app gave me a result. Also whatsapp is banning with delay so the registration could be a trigger.

**Comment by schoerg** [comment]
And could you login with the password?

**Comment by sowerkoku** [comment]
hope so. that's the idea.

**Comment by rami-dabain** [comment]
i could extract the passwird from the app, but when using it with yowsup it
says auth Required. its Over guys. time for telegram

**Comment by MrGlue** [comment]
Tried it too. Unsuccessfully. @schoerg was right.

**Comment by schoerg** [comment]
@rami-dabain you can still register with yowsup-cli. 
Telegram has a really good API, the problem is that next to no one is using it.

**Comment by sowerkoku** [comment]
even it works perfect for me yowsup.
I have two bot running about 3 months ago and recorded a new number with yowsup 3 days ago.

**Comment by davidsantanacosta** [comment]
@rami-dabain  quitting is for cowards

**Comment by falbani** [comment]
Please stop with all the useless "Telegram" comments! This is not a Telegram project! Also the user base of Telegram is ridiculously small to even dare to comparing it to the Whatsapp one.

We all know Whatsapp is closed source, obfuscated, closed protocol, etc; We all know Whatsapp is an spy tool now owned by Facebook corp, we all know all the messages are stored forever in their servers and they have a profile on everyone who uses it. But we can NOT change the fact that marketing has pushed Whatsapp to the masses, billions of users use it everyday worldwide, in that masses are our bosses, clients, friends, enemies, family, girlfriend, boyfriend, whatever... ergo we are FORCED to use this crap.

With that being said we should focus all our time and resource in reverse engineering their network protocol and deobfuscate the android app to mimic it exactly.

**Comment by biji** [comment]
Just some idea , maybe we can use another approach for library, by remoting WhatsApp web using node js

**Comment by assegaf** [comment]
@falbani  how you know that its spy purpose, the algoritm is used for end2end encryption is opensource, you can read the code that password/private key is never exposed to server.

so fbook only know u, number U send something (txt/media/etc) to x Number X,
and off course probably ip, but I think ip is general.

**Comment by poma** [comment]
@assegaf metadata is usually more valuable than content. Just look at NSA leaks - most of their work is centered around metadata. So "u, number U send something (txt/media/etc) to x Number X" is all they want to know.

**Comment by schoerg** [comment]
Also, metadata can be much easier automatically analyzed.

**Comment by cayasso** [comment]
I agree with @falbani and @biji maybe its time to explore another approach, I know its not easy but it may worth the effort, what do you guys think?

**Comment by schoerg** [comment]
I don't know what you guys are doing, but I haven't been banned with the latest yowsup release. I updated the WA version one time, but thats about it.

I have been using a "real" number, not a Voip or contractless card. I am not even trying to "humanize" the  yowsup client by adding random delay or setting the "... typing" stuff, because I didn't care, and it doesn't seem to be required anyway.

**Comment by sowerkoku** [comment]
humanize the system has no utility rather than for the end user.
just double check, so the response time is much faster.
humanized my system, my client liked a lot. but for personal use what I consider totally unnecessary.

**Comment by hsteffens95** [comment]
hi guys is there anything that solved the problem with the Login Fail because of not-autorized.

**Comment by berryelectronics** [comment]
g'day! Joining to the conversation. didnt had time to go through all of the comments, but still:

has anyone tried to hit up some of the services like whatsbroadcast and asked them how they manage that?
